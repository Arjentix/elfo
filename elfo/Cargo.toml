[package]
name = "elfo"
version = "0.1.1"
description = "Your next actor system"
keywords = ["actor", "distributed", "tokio"]
authors = ["Paul Loyd <pavelko95@gmail.com>"]
repository = "https://github.com/loyd/elfo"
license = "MIT"
readme = "README.md"
edition = "2018"

[[test]]
name = "test"
path = "examples/test.rs"

[features]
test-util = ["elfo-test", "elfo-core/test-util"]

[dependencies]
elfo-core = { version = "0.1", path = "../elfo-core" }
elfo-macros = { version = "0.1", path = "../elfo-macros" }
elfo-test = { version = "0.1", path = "../elfo-test", optional = true }
elfo-configurer = { version = "0.1", path = "../elfo-configurer", optional = true }

[dev-dependencies]
anyhow = "1.0.38"
tokio = { version = "1", features = ["full"] }
tracing = "0.1.25"
tracing-subscriber = "0.2.15"
serde = { version = "1.0.120", features = ["derive"] }
toml = "0.5.8"
humantime-serde = "1"
