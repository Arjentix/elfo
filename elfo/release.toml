pre-release-replacements = [
    {prerelease=true, file="../README.md", search="^elfo =(.*?)\".+?\"", replace="{{crate_name}} =${1}\"{{version}}\""},
    {prerelease=true, file="../README.md", search="docs.rs/elfo/.*", replace="docs.rs/elfo/{{version}}/elfo"},
    {prerelease=true, file="../CHANGELOG.md", search="Unreleased", replace="{{version}}"},
    {prerelease=true, file="../CHANGELOG.md", search="\\.\\.\\.HEAD", replace="...{{tag_name}}", exactly=1},
    {prerelease=true, file="../CHANGELOG.md", search="ReleaseDate", replace="{{date}}"},
    {prerelease=true, file="../CHANGELOG.md", search="<!-- next-header -->", replace="<!-- next-header -->\n\n## [Unreleased] - ReleaseDate", exactly=1},
    {prerelease=true, file="../CHANGELOG.md", search="<!-- next-url -->", replace="<!-- next-url -->\n[Unreleased]: https://github.com/elfo-rs/elfo/compare/{{tag_name}}...HEAD", exactly=1},
]
