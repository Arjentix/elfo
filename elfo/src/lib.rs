#![warn(rust_2018_idioms, unreachable_pub)]

pub use elfo_core::*;
pub use elfo_macros::{message, msg};

#[cfg(feature = "elfo-configurer")]
pub use elfo_configurer as configurer;

pub mod prelude {
    pub use super::{assert_msg, assert_msg_eq, message, msg, ActorGroup, Context, Schema};
}

#[deprecated(since = "0.1.1")]
pub mod actors {
    #[cfg(feature = "elfo-configurer")]
    pub use super::configurer;
}

#[cfg(feature = "test-util")]
pub use elfo_test as test;
