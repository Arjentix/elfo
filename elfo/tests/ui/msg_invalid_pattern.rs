use elfo::{msg, Envelope};

// TODO: check more invalid patterns.
fn test(envelope: Envelope) {
    msg!(match envelope {
        foo!() => {}
        "liternal" => {}
        10..20 => {}
        (A | B) => {}
        (SomeRequest, token, extra) => {}
        (SomeRequest, 20) => {}
    });
}

fn main() {}
