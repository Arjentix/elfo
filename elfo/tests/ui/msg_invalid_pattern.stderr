error: macros in pattern position are forbidden
 --> tests/ui/msg_invalid_pattern.rs:6:9
  |
6 |         foo!() => {}
  |         ^^^

error: literal patterns are forbidden
 --> tests/ui/msg_invalid_pattern.rs:7:9
  |
7 |         "liternal" => {}
  |         ^^^^^^^^^^

error: range patterns are forbidden
 --> tests/ui/msg_invalid_pattern.rs:8:9
  |
8 |         10..20 => {}
  |         ^^

error: parenthesized patterns are forbidden
 --> tests/ui/msg_invalid_pattern.rs:9:9
  |
9 |         (A | B) => {}
  |         ^^^^^^^

error: invalid request pattern
  --> tests/ui/msg_invalid_pattern.rs:10:9
   |
10 |         (SomeRequest, token, extra) => {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: token must be identifier
  --> tests/ui/msg_invalid_pattern.rs:11:9
   |
11 |         (SomeRequest, 20) => {}
   |         ^^^^^^^^^^^^^^^^^
